### ISSUE TITLE
Segmentation fault compiling Lib Bpg

### ISSUE THREAD

**Issue by yahsaves** [kripken/emscripten#5969]
I'm trying to compile Fabrice Bellard's [Lib Bpg](https://bellard.org/bpg/) project. 

I get the program to build into it's .js and .wasm files, however when I try to load them and run them in the browser I get a warning that says:

falling back to ArrayBuffer instantiation

Then I get a critical error that says:

Uncaught RuntimeError: memory access out of bounds

I tried compiling with -s SAFE_HEAP=1. That changes the error to read:

uncaught exception: abort("segmentation fault")

I am using emscripten version 1.37.26, and there are no errors on my system when I run the sanity checks. 

What I am trying to figure out is Mr. Bellard had compiled his project into asm.js successfully about a year ago, so I know the code must work, I just wanted to see if I could get better performance with wasm. 

Since my build is failing I tried to force emscripten to just compile to asm.js, and not to wasm. However I couldn't find any updated flags that would do so. I tried this "-s WASM=0" but it still built wasm code.

Any help would be much appreciated!

---Edit
    I should mention that by using the standard "make" command on my system I can successfully compile the fully working executables of the project. Also I am using "cmmake make" to build the project.

---Edit V2
   As a test I downloaded an older version of emscripten (1.36.11) and it compiled into normal asm.js and worked successfully. It appears only wasm is giving problems.

**Comment by kripken** [comment]
Some tips for debugging when asm.js works but wasm fails: https://github.com/kripken/emscripten/wiki/WebAssembly#debugging

**Comment by yahsaves** [comment]
Thank you for the fast response! 

I read over the link you sent, but unfortunately the only thing I could get working was when I compiled the project with the -s "BINARYEN_METHOD='asmjs,native-wasm'" flag.

Meaning it would default to Asm.js first. When I tried it the other way around (putting wasm first) I got this error:

Assertion failed: on the web, we need the wasm binary to be preloaded and set on Module['wasmBinary']. emcc.py will do that for you when generating HTML (but not JS)

On the flip side I did see [this link](https://github.com/kripken/emscripten/wiki/New-WebAssembly-Backend) talking about a new wasm backend compiler. I understand that it is experimental, but do you think it has any chance of compiling the program correctly? 

Thank you for  your help!

**Comment by saschanaz** [comment]
Did you set `Module.wasmBinary` as the assetion message says?

**Comment by yahsaves** [comment]
Well I have no idea what I did, but coming back to this project after a few days I tried to compile to wasm with these settings:   

-s EXTRA_EXPORTED_RUNTIME_METHODS='["ccall", "cwrap"]' -s WASM=1 -s NO_FILESYSTEM=1  -s "BINARYEN_METHOD='native-wasm'"

And for whatever reason it's working. I have a wasm enabled build that actually works. I tried all these settings nearly 50 times before, but I'm not complaining. Thank you everyone so much for your help!
